Use GitHub Actions and Rust Edition 2024

p14c31355 opened the pull request:
This should resolve #12 and #13.

p14c31355 committed:
Replace Travis and AppVeyor with GitHub Actions

Closes #13.

p14c31355 committed:
Use Rust edition 2024

This patch was created by running:

    cargo +nightly fix --edition --allow-dirty

Closes #12.

klingtnet commented:
@YoshitakaNaraoka In general your commits look fine, however I'd like to adapt the commit messages a bit and rewrite the GitHub Actions so they test against different Rust versions. Are you okay with me rewriting your commits?

klingtnet commented:
@YoshitakaNaraoka I added my changes as individual fixup commits. If you're fine with them then I'm going to squash and merge this PR.

p14c31355 commented:
@klingtnet Thank you & okay !!
I pray that this repository will be ready for the next era with this modification!

klingtnet committed:
Fix clearing the buffer

Clippy complained about the ineffective `count()`, whose whole purpose
was to make sure that the iterator gets consumed. However, we can do
this more properly by calling `.for_each()` on the iterator.

klingtnet reviewed (approved):
LGTM.
